crates/python/src/lib.rs

```diff
@@ -10,24 +11,31 @@ create_exception!(qcs, InvalidConfigError, PyRuntimeError);
 create_exception!(qcs, ExecutionError, PyRuntimeError);
 create_exception!(qcs, TranslationError, PyRuntimeError);
 create_exception!(qcs, CompilationError, PyRuntimeError);
+create_exception!(qcs, RewriteArithmeticError, PyRuntimeError);
 
 #[pyfunction]
 fn compile(py: Python<'_>, quil: String, target_device: String) -> PyResult<&PyAny> {
     let target_device: TargetDevice = serde_json::from_str(&target_device)
         .map_err(|e| CompilationError::new_err(e.to_string()))?;
     pyo3_asyncio::tokio::future_into_py(py, async move {
-        let config = Configuration::load()
+        // TODO This doesn't need to be async.
+        let config = ClientConfiguration::load()
             .await
             .map_err(|e| InvalidConfigError::new_err(e.to_string()))?;
-        let result = api::compile(&quil, target_device, &config)
+        let client = QcsClient::with_config(config);
+        let result = api::compile(&quil, target_device, &client)
             .map_err(|e| CompilationError::new_err(e.to_string()))?;
         Ok(Python::with_gil(|_py| result))
     })
 }
 
 #[pyfunction]
 fn rewrite_arithmetic(py: Python<'_>, native_quil: String) -> PyResult<PyObject> {
-    let result = api::rewrite_arithmetic(&native_quil).map_err(TranslationError::new_err)?;
+    let native_program = NativeQuil::assume_native_quil(native_quil)
+        .try_into()
+        .map_err(TranslationError::new_err)?;
+    let result = api::rewrite_arithmetic(native_program)
+        .map_err(|e| RewriteArithmeticError::new_err(e.to_string()))?;
```

Comment:
I believe this takes anything implementing `Display`

```suggestion
        .map_err(RewriteArithmeticError::new_err)?;
```

Reply:
Doesn't seem to work out, but maybe I'm missing something. LMK if you have success.